### ISSUE TITLE
Reading compressed FITS files is slow ?

### ISSUE THREAD

**Issue by saimn** [astropy/astropy#7776]
From Astropy performance survey:

> fitsio; tile compressed files i/o of cutout is very slow compared with Erin Sheldons cfitsio based fitsio package; I can get someone in my group to run some timing tests. From @richardgmcmahon

> FITS I/O, in comparison to cfits io loading is slow especially for RICE compressed images. From @Cadair 

Note that the reading of tiled compressed files in Astropy is actually done with cfitsio (and this is the only reason why, to my knowledge, cfitsio is bundled in Astropy). 
So it would be great to have some minimal exemple with performance timings to see how it is slow, and if it can be improved.

**Comment by Cadair** [comment]
I would need to rerun my tests in a bit more of a controlled manner. The very limited profiling I did seemed to indicate there was a *lot* of overhead in the closing of the file. (exiting the context manager).

**Comment by astrofrog** [comment]
@saimn @Cadair - I wonder if it would be worth closing this or maybe checking first if @Cadair's point about closing the file is still true. But with the ``section`` API the timings for reading compressed files is not way worse than the fitsio package.

**Comment by saimn** [comment]
Agreed that the first item about tile compression is fixed, not sure about the second one from @Cadair.
